src/tools/diagnostics/analyze_resource/analyze_resource_cloud_vendor.js

```diff
@@ -0,0 +1,76 @@
+/* Copyright (C) 2023 NooBaa */
+'use strict';
+
+const fs = require('fs');
+const http_utils = require('../../../util/http_utils');
+const SensitiveString = require('../../../util/sensitive_string');
+const AnalyzeAws = require('./analyze_resource_aws');
+const AnalyzeGcp = require('./analyze_resource_gcp');
+const AnalyzeAzure = require('./analyze_resource_azure');
+
+const secret_path = '/etc/cloud-credentials';
+
+function get_cloud_vendor(resource_type, connection_basic_details) {
+    let cloud_vendor;
+    const credentials = get_credentials(resource_type);
+
+    switch (resource_type) {
+        case 'aws-s3':
+            cloud_vendor = new AnalyzeAws(credentials.access_key, credentials.secret_access_key,
+                connection_basic_details.endpoint, connection_basic_details.signature_version,
+                http_utils.get_default_agent(connection_basic_details.endpoint));
+            break;
+        case 's3-compatible':
+            cloud_vendor = new AnalyzeAws(credentials.access_key, credentials.secret_access_key,
+                connection_basic_details.endpoint, connection_basic_details.signature_version,
+                http_utils.get_unsecured_agent(connection_basic_details.endpoint));
+            break;
+        case 'ibm-cos':
+            cloud_vendor = new AnalyzeAws(credentials.access_key, credentials.secret_access_key,
+                connection_basic_details.endpoint, connection_basic_details.signature_version,
+                http_utils.get_unsecured_agent(connection_basic_details.endpoint));
+            break;
+        case 'azure-blob':
+            cloud_vendor = new AnalyzeAzure(credentials.access_key, credentials.secret_access_key, // Azure storage account name is stored as the access key
+                connection_basic_details.endpoint);
+            break;
+        case 'google-cloud-storage':
+            cloud_vendor = new AnalyzeGcp(credentials.private_key_json);
+            break;
+        default:
+            throw new Error(`Could not create cloud vendor client of ${resource_type}`);
+    }
+    return cloud_vendor;
```

Comment:
cloud_vendor should be initialized with some null value or something. What if we come to default case and from there we return cloud_vendor which has not value.
although I  am not sure what happens to uninitialized variable in JS/

Reply:
`cloud_vendor` variable was declared and its initial value is undefined (you can find it in the [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let)): 

> The variable is initialized with a value when execution reaches the line of code where it was declared. If no initial value was specified with the variable declaration, it will be initialized with a value of undefined.
A simple 2 lines program illustrates it:
```
let foo;
console.log(foo);
```
output:
undefined

If we get to this function with a `resource_type` with a resource type that hits the default case we will throw an error and would not get to the return statement. we call this function from the main, so we will hit the catch block there.

Reply:
Yeh, That is okay. I thought "undefined" can not be used as "null check" but I saw that it gives the same result in JS.